crates/testing/src/test_task.rs

```diff
@@ -97,13 +97,18 @@ impl<S: TestTaskState + Send + 'static> TestTask<S> {
                     messages.push(receiver.recv());
                 }
 
-                if let Ok((Ok(input), id, _)) =
-                    async_timeout(Duration::from_millis(50), select_all(messages)).await
-                {
-                    let _ = S::handle_event(&mut self.state, (input, id))
-                        .await
-                        .inspect_err(|e| tracing::error!("{e}"));
-                }
+                match async_timeout(Duration::from_millis(2500), select_all(messages)).await {
+                    Ok((Ok(input), id, _)) => {
+                        let _ = S::handle_event(&mut self.state, (input, id))
+                            .await
+                            .inspect_err(|e| tracing::error!("{e}"));
+                    }
+                    Ok((Err(e), _id, _)) => {
+                        error!("Error from one channel in test task {:?}", e);
```

Comment:
not sure if this would be better, but a thought: could we match and drop the receiver here? e.g. `Ok((Err(RecvError::Closed), id, _)) => { self.receivers.remove(id) }`

(this might require also breaking at the start if `self.receivers.is_empty()`)

Reply:
yeah I want to do that, but it would change the idx of the receiver and might lead to some weird behaviours.  I plan to follow up with this with a few more cleanups and will store the receivers with their node_id somehow in this task (or replace with empty or new receivers if possible)